app/views/admin/order_cycles/checkout_options.html.haml

```diff
@@ -15,66 +15,58 @@
-        %tr
-          %td= distributor.name
-          %td
-            - shipping_methods.each do |shipping_method|
-              %p
-                %label
-                  = check_box_tag "order_cycle[selected_shipping_method_ids][]",
-                      shipping_method.id, @order_cycle.shipping_methods.include?(shipping_method),
-                      id: "order_cycle_selected_shipping_method_ids_#{shipping_method.id}"
-                  = shipping_method.name
-            - distributor.shipping_methods.backend.each do |shipping_method|
-              %label.disabled
-                = check_box_tag nil, nil, false, disabled: true
-                = shipping_method.name
-                = "(#{t('.back_end')})"
-            - if distributor.shipping_methods.frontend.none?
-              %p
-                = t('.no_shipping_methods')
-          %td
-            - if distributor.payment_methods.available(:both).any?
-              %ul
-                - payment_methods.each do |payment_method|
-                  %li= payment_method.name
-            - else
-              %p
-                = t('.no_payment_methods')
-      - if shared_payment_methods.any? || shared_shipping_methods.any?
-        %tr
-          %td= t('.shared')
-          %td
-            - shared_shipping_methods.each do |shared_shipping_method|
-              %p
-                %label
-                  = check_box_tag "order_cycle[selected_shipping_method_ids][]",
-                      shared_shipping_method.id, @order_cycle.shipping_methods.include?(shared_shipping_method),
-                      id: "order_cycle_selected_shipping_method_ids_#{shared_shipping_method.id}"
-                  = shared_shipping_method.name
-              %p
-                &mdash;
-                %em
-                  = shared_shipping_method.distributors.where(id: @order_cycle.distributor_ids).map(&:name).join(", ")
-          %td
-            - if shared_payment_methods.any?
-              %ul
-                - shared_payment_methods.each do |shared_payment_method|
-                  %li
-                    = shared_payment_method.name
-                    %p
+  .row
+    .three.columns
+      &nbsp;
+    .ten.columns
+      %table.checkout-options
+        %thead
+          %tr
+            %th{ colspan: 2 }= t('.shipping_methods')
+          %tr{ "data-controller": "select-all" }
+            %td.text-center
+              %label
+                = check_box_tag "bla", nil, shipping_methods == @order_cycle.shipping_methods, { "data-action": "change->select-all#toggleAll", "data-select-all-target": "all" }
+                = "Select all"
+            %td
+              - if shipping_methods.any?
+                - shipping_methods.each do |shipping_method|
+                  %p
+                    %label{ class: ("disabled" if shipping_method.backend?) }
```

Comment:
Do we need the class? Is the attribute not enough?

Reply:
I think we do. The `.disabled` class on the label adds the grey text and makes it look less clickable e.g.

![disabled-text](https://user-images.githubusercontent.com/101088/181722035-7c479927-0d89-447a-ab50-6f77b3ab4f16.png)